info-v4-resources.go

```diff
@@ -47,29 +47,52 @@ type PaginatedPoolsResponse struct {
 
 // PaginatedNodesResponse represents a paginated response for nodes
 type PaginatedNodesResponse struct {
-	Results      []NodeResource `json:"results" msg:"r,omitempty"`
-	Count        int            `json:"count" msg:"c"`
-	Total        int            `json:"total" msg:"t"`
-	Offset       int            `json:"offset" msg:"o"`
-	Sort         string         `json:"sort" msg:"s"`
-	SortReversed bool           `json:"sortReversed" msg:"sr"`
+	Results      []NodeResource    `json:"results" msg:"r,omitempty"`
+	Summary      NodesQuerySummary `json:"summary" msg:"sum"`
+	Count        int               `json:"count" msg:"c"`
+	Total        int               `json:"total" msg:"t"`
+	Offset       int               `json:"offset" msg:"o"`
+	Sort         string            `json:"sort" msg:"s"`
+	SortReversed bool              `json:"sortReversed" msg:"sr"`
+}
+
+// NodesQuerySummary contains summary statistics for all nodes in the cluster
+type NodesQuerySummary struct {
+	Offline      int `json:"Offline" msg:"off"`
+	Initializing int `json:"Initializing" msg:"ini"`
+	Online       int `json:"Online" msg:"on"`
+	Unknown      int `json:"Unknown" msg:"un"`
 }
 
 // PaginatedDrivesResponse represents a paginated response for drives
 type PaginatedDrivesResponse struct {
-	Results      []DriveResource `json:"results" msg:"r,omitempty"`
-	Count        int             `json:"count" msg:"c"`
-	Total        int             `json:"total" msg:"t"`
-	Offset       int             `json:"offset" msg:"o"`
-	Sort         string          `json:"sort" msg:"s"`
-	SortReversed bool            `json:"sortReversed" msg:"sr"`
+	Results      []DriveResource    `json:"results" msg:"r,omitempty"`
+	Summary      DrivesQuerySummary `json:"summary" msg:"sum"`
+	Count        int                `json:"count" msg:"c"`
+	Total        int                `json:"total" msg:"t"`
+	Offset       int                `json:"offset" msg:"o"`
+	Sort         string             `json:"sort" msg:"s"`
+	SortReversed bool               `json:"sortReversed" msg:"sr"`
 
 	// Aggregated are the metrics aggregated for all filtered drives,
 	// not just the results.
 	// Always returned, though day metrics are only available if the option is set.
 	Aggregated DiskMetric `json:"aggregated,omitempty" msg:"m,omitempty"`
 }
 
+// DrivesQuerySummary contains a summary for all drives, ignoring pagination and query limits.
```

Comment:
just a thought here but since DrivesQuerySummary and Aggregated ignore pagination I was wondering if we should create another returning struct that contains both the PaginatedDrivesResponse and this new ones. Just to encapsulate how they work and make it self explanatory.

type DrivesQuerySummary struct {
    PaginatedResults PaginatedDrivesResponse
	Summary DrivesQuerySummary
    Aggregated DiskMetric `json:"aggregated,omitempty" msg:"m,omitempty"`
}


same for Nodes.